cpp/include/raft/distance/detail/compress_to_bits.cuh

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <raft/util/cuda_utils.cuh>
+#include <raft/util/device_atomics.cuh>
+
+namespace raft {
+namespace distance {
+namespace detail {
+
+template <typename T = uint64_t, typename = std::enable_if_t<std::is_integral<T>::value>>
+__global__ void compress_to_bits_naive(const bool* in, int in_rows, int in_cols, T* out)
```

Comment:
So this accepts a boolean adjacency matrix and compresses the bytes to a `uint64_t` bitset? This seems super useful. Any reason we shouldn't add this to `raft/util/mask.cuh` or even `raft/sparse/mask.cuh`?

Either way, I would suggest putting a non-`__global__` helper function in this file that can wrap this kernel to provide the needed geometry and kernel settings.